### PR TITLE
Add tests for -articles-tags.js-

### PR DESCRIPTION
**Pull request by pointbar**
@davidbgk & @vinyll - some questions to finalize : 
- I have no idea to the policy for JS libraries installation - ex. npm or /vendors
- I think, it's better if Travis plays the tests, no ?
- For instant, I use JSM, perhaps you prefer Firefox headless or something like Selenium ?

### CODE REVIEW

File: cafebabel/static/js/articles-tags.js [hunk 1]
```diff
@@ -1,70 +1,82 @@
...
-    return request(
-      `/article/tag/suggest/?language=${this.language}&terms=${submission}`
-    ).catch(console.error.bind(console))
+    const url = `/article/tag/suggest/?language=${
+      this._language
+    }&terms=${submission}`
+    return fetch(url)
+      .then(handleResponse)
+      .catch(error => {
+        const e = new Error(`${error.message} ${url}`)
```
Comment (davidbgk) [inline cafebabel/static/js/articles-tags.js]: You should not have to do that as it is [already handled](https://github.com/cafebabel/cafebabel.com/blob/master/cafebabel/static/js/utils.js#L8) by the picolib we use, `error.url` should already be there 😉

**Comment by vinyll** [comment]
> I have no idea to the policy for JS libraries installation - ex. npm or /vendors

If you can have one stable final single file for usage, it feels like less of a fuzz than using npm. Unsure about side effects though.

> I think, it's better if Travis plays the tests, no ?

I guess so too. We'll probably _never_ run the tests ourselves.

> For instant, I use JSM, perhaps you prefer Firefox headless or something like Selenium ?

No experience hence no strong opinion.

File: cafebabel/static/js/articles-tags.js [hunk 2]
```diff
@@ -1,42 +1,47 @@
...
-    return languages.options[languages.selectedIndex].value
-  }
-  get list() {
-    return this.context.querySelector('.tags-list')
+  get context() {
+    return document.querySelector('.tags')
   }
   get removeButtons() {
-    return this.list.querySelectorAll('li button')
+    return this._list.querySelectorAll('li button')
```
Comment (vinyll) [inline cafebabel/static/js/articles-tags.js]: https://gist.github.com/DavidBruant/1016007

File: cafebabel/static/js/articles-tags.js [hunk 3]
```diff
@@ -1,42 +1,47 @@
...
     return this.context.querySelector('input[name=tag-new]')
   }
-  get buttonAdd() {
+  get addButton() {
     return this.context.querySelector('button.add')
   }
   get suggestions() {
+    return this._suggestionsContainer.querySelectorAll('li')
+  }
+  get _languages() {
```
Comment (vinyll) [inline cafebabel/static/js/articles-tags.js]: Not a big fan of the _wrap-everything_ approach. It does abstract the dom in a good way and would make the code more generic, but it duplicates the lines and the code is not generic in the end anyways.

It does make sense for actual verbe methods (`addNewTag`, `removeTag`) though

File: cafebabel/templates/articles/macros.html [hunk 4]
```diff
@@ -22,8 +22,8 @@
     <ul class=tags-list>
       {% if article and article.tags %}
         {% for tag in article.tags -%}
-          <li class=saved>
-            {{ tag.name }}{# Avoid any whitespace between the name and the input.
+          <li class=saved>{# Avoid any whitespace before the name and the input.
```
Comment (vinyll) [inline cafebabel/templates/articles/macros.html]: is it an issue also before?

File: cafebabel/static/js/articles-tags.js [hunk 5]
```diff
@@ -1,42 +1,47 @@
...
+    return Array.from(this._tags).map(({ value }) => value)
+  }
+  get _list() {
+    return this.context.querySelector('.tags-list')
+  }
+  get _suggestionsContainer() {
     return this.context.querySelector('#tags-suggestions')
   }
 
   addNewTag(submission) {
```
Comment (vinyll) [inline cafebabel/static/js/articles-tags.js]: Why `addNewTag` and not `addTag` while there is `removeTag`?

**Review by vinyll — COMMENTED** [review]
Just a quick pre-review.
This one will be time demanding for reading either the entire changes at once or understanding per commit.
Not sure how to help 🤔

File: cafebabel/templates/articles/macros.html (hunk 4 above)
Comment (pointbar) [reply cafebabel/templates/articles/macros.html]: I try without, the tests pass and it seems works. So thx

**Comment by vinyll** [comment]
screen shot 2018-02-06 at 5 23 32 pm without even the actual code of the tags and it's usage is triggering an alert to me.
This means we should maintain over 1200 (1500?) lines of code for an input tags field.

Did we ever consider something in Vanilla that is already tested and maintained (like [Choices](https://github.com/jshjohnson/Choices)), [TaggleJS](https://github.com/okcoker/taggle.js) or other libraries?

Not sure how I can help on this. @davidbgk, any thought over all this?

**Comment by davidbgk** [comment]
Reopen when/if we keep the tag custom lib and want to test it.

**Pull request closed without merging**